doc/developers.rst

```diff
@@ -66,37 +68,82 @@ Install Autosklearn
-   
-   3. Install xgboost and autosklearn
+      Use your package manager to install gcc (necessary for xgboost)
    
       .. code-block:: console
       
-         $ export CC=/path/to/installed/gcc; export CXX=/path/to/installed/g++; pip install xgboost
-         $ pip install autosklearn
+         (venv) $ brew install gcc # (or apt-get)
+
+Install all the packages and commit hooks
+   When the project is installed through poetry both project requirements and development requirements are installed. Install commit-hooks using the `pre-commit`_ utility.
+
+   .. _pre-commit: https://pre-commit.com/
+
+   .. code-block:: console
+   
+      (venv) $ poetry install -v
+      (venv) $ poetry run pre-commit install
 
 Making sure everything works
    1. Run pytest to make sure you're good to go
    
       .. code-block:: console
       
-         $ pytest
+         (venv) $ poetry run pytest
    
    2. Run tox to run in supported python versions (optional)
    
       .. code-block:: console
       
-         $ tox # supply the -r flag if you changed the dependencies
+         (venv)$ poetry run tox -r # supply the -r flag if you changed the dependencies
+
+   3. Run make html in foreshadow/doc to build the documentation (optional)
+   
+      .. code-block:: console
+      
+         (venv) $ poetry run make html
    
    If all the tests pass you're all set up!
 
+Suggested development work flow
+   1. Create a branch off of development to contain your change
+   
+      .. code-block:: console
+      
+         (venv) $ git checkout development
+         (venv) $ git checkout -b {your_feature}
+
+   2. Run pytest and pre-commit while developing
+      This will help ensure you haven't horrifically broken something while adding your feature and will help you catch bugs as you develop. Pre-commit will help make sure that your formatting is pristine before create a pull request.
+   
+      .. code-block:: console
+      
+         $ poetry run pytest
+         $ poetry run pre-commit run --all-files
+
+   3. Run tox to test your changes across versions
+      Make sure to add test cases for your change in the appropriate folder in foreshadow/tests and run tox to test your project across python 3.5 and 3.6
+
+      .. code-block:: console
+      
+         $ poetry run tox
+
+   4. Submit a pull request
+      This can be tricky if you have cloned the project instead of forking it but no worries the fix is simple. First go to the project page and **fork it there**. Then do the following.
+
+      .. code-block:: console
+      
+         (venv) $ git remote add upstream https://github.com/georgianpartners/foreshadow.git
+         (venv) $ git remote set-url origin https://github.com/{YOUR_USERNAME}/foreshadow.git
+         (venv) $ git push origin {your_feature}
+   
+      Now you can go to the project on your github page and submit a pull request to the main project. Note, make sure to submit the pull request against the development branch.
```

Comment:
Should the "note" be separated into an actual note block? It seems important enough to warrant in my opinion.

Reply:
Sure I can make that change.